### PR TITLE
Exception handling initial commit

### PR DESCRIPTION
**Pull request by rswijesena**

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/exception/ExceptionCodes.java [hunk 1]
```diff
@@ -26,14 +26,18 @@
  */
 public enum ExceptionCodes {
 
-    API_ALREADY_EXISTS(900300, "The API already exists.");
+    API_ALREADY_EXISTS(900300, "The API already exists.", 400),
+    APIMGT_DAO_EXCEPTION(900301, "Something wrong in DAO layer.", 500),
+    APIMGT_LIFECYCLE_EXCEPTION(900301, "Life cycle exception occured", 500);
```
Comment (nuwand) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/exception/ExceptionCodes.java]: Duplicate error code 900301

File: components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/exception/ExceptionCodes.java [hunk 2]
```diff
@@ -26,14 +26,18 @@
  */
 public enum ExceptionCodes {
 
-    API_ALREADY_EXISTS(900300, "The API already exists.");
+    API_ALREADY_EXISTS(900300, "The API already exists.", 400),
```
Comment (nuwand) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/exception/ExceptionCodes.java]: Its better to explain in a comment what these parameters are (unique error code, error message, corresponding http status code to be returned)

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/impl/ApisApiServiceImpl.java [hunk 3]
```diff
@@ -387,10 +387,15 @@ public Response apisPost(APIDTO body
         } catch (APIManagementException e) {
             String errorMessage = "Error while adding new API : " + body.getProvider() + "-" +
                     body.getName() + "-" + body.getVersion();
+            HashMap<String, String> paramList = new HashMap<String, String>();
+
+            paramList.put("API_NAME", body.getProvider());
```
Comment (nuwand) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/impl/ApisApiServiceImpl.java]: Use constants

**Comment by rswijesena** [comment]
I am closing this pull request. Will send new one with more the changes that discussed in the code review.

**Pull request closed without merging**